### PR TITLE
Sort and Search functionality added to Table component

### PR DESCRIPTION
**Pull request by cbunnydev**
Table component is updated to be sorted by column
Table component is updated to be filtered by query

### CODE REVIEW

**Review by hachiojidev — CHANGES_REQUESTED** [review]
Is this supposed to be a draft? There are many linting errors that need to be fixed.

**Comment by cbunnydev** [comment]
> 
> 
> Is this supposed to be a draft? There are many linting errors that need to be fixed.

No, this was fully functional, just needed to fix the linting errors.

File: src/components/Table/hooks.tsx [hunk 1]
```diff
@@ -214,22 +399,28 @@ export const useTable = <T extends DataType>(
...
-      nextPage: () => {
-        // nextPage feature
-      },
-      prevPage: () => {
-        // prevPage feature
-      },
+      // eslint-disable-next-line @typescript-eslint/no-empty-function
+      nextPage: () => {},
+      // eslint-disable-next-line @typescript-eslint/no-empty-function
+      prevPage: () => {},
```
Comment (hachiojidev) [inline src/components/Table/hooks.tsx:405]: Use Lodash's `noop`

File: src/components/Table/hooks.tsx [hunk 2]
```diff
@@ -66,99 +73,275 @@ const getColumnsByName = <T extends DataType>(columns: ColumnType<T>[]): ColumnB
...
+  value: any,
+  render: (({ value: val, row }: { value: any; row: T }) => ReactNode) | undefined,
+  row: T
+) => {
+  return render ? () => render({ row, value }) : () => value;
+};
+
+const makeHeaderRender = (
+  label: string,
+  render: HeaderRenderType | undefined
```
Comment (hachiojidev) [inline src/components/Table/hooks.tsx:332]: This should be `render?: HeaderRenderType`

**Review by hachiojidev — CHANGES_REQUESTED** [review]
Please keep this in a draft PR until the linting and prettier has been fixed. I recommend having Prettier format your file on each save.

**Comment by RabbitDoge** [comment]
There is a lot of formatting problems. It looks like your prettier doesn't use the project config. Run `yarn format` to see the difference.

**Review by hachiojidev — APPROVED** [review]

**Pull request merged**